### PR TITLE
[Page Content Fetch] added initial setup to have page contents

### PR DESCRIPTION
**Pull request by stepsen89**
- [x] uses getServerSideProps
- [x] uses helper method for api calls to pass in file name (for now, will be path later)
- [x] copies over home.json from API contract  
- [x] temp error page for setup

### CODE REVIEW

File: public/home.json [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        "speaker": "John Smith",
+        "description": "Join us for a discussion of this essential guide to writing clear and maintainable code!\n",
+        "images": [
+          {
+            "path": "https://cloudprovider.com/image.png",
+            "alt": "There are two women talking during a productive mentoring session",
+            "type": "desktop"
+          }
+        ],
+        "link": "https://www.linkedin.com"
```
Comment (dricazenck) [inline public/home.json:62]: the link will look like this: 

"link": {
          "label": "Check the meetup event",
          "uri": "https://meetup.com/event/event1",
          "title": ""
        }

File: src/utils/types.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+type WCCEvent = {
+  title: string;
+  type: string;
+  host: string;
+  speaker: string;
+  date: string;
+  endDate: number;
+  description: string;
+  link: string;
```
Comment (dricazenck) [inline src/utils/types.ts:21]: The link will be a bit different, because will contain label, uri.

File: src/pages/404.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+};
+
+const styles = {
+  container: {
+    display: 'flex',
+    flexDirection: 'column' as 'column',
+    alignItems: 'center',
+    justifyContent: 'center',
+    height: '80vh',
+    textAlign: 'center' as 'center',
```
Comment (dricazenck) [inline src/pages/404.tsx:23]: dummy question, why did you did this?  `'center' as 'center'`

File: src/utils/types.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  type?: string;
+};
+
+type Programme = {
+  name: string;
+  link: string;
+  icon: string;
+};
+
+type WCCEvent = {
```
Comment (dricazenck) [inline src/utils/types.ts:13]: what do you think about `CommunityEvent` or just `Event`?

Comment (stepsen89) [reply src/utils/types.ts]: updated - went with `Event` now 😅

File: src/pages/404.tsx (hunk 3 above)
Comment (stepsen89) [reply src/pages/404.tsx:23]: that was just a TypeScript error, didn't spend much time changing the TS setup as this is a dummy component (we currently don't have a design for this). But it basically is to assert the type of the string to FlexDirection (and TextAlign). using MUI this shouldn't occur

File: public/home.json [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    "link": "https://www.linkedin.com",
+    "images": [
+      {
+        "path": "https://cloudprovider.com/image.png",
+        "alt": "There are two women talking during a productive mentoring session",
+        "type": "desktop"
+      }
+    ]
+  }
+}
```
Comment (dricazenck) [inline public/home.json:82]: If you want to update with this one: 

```
{
  "heroSection": {
    "title": "Women Coding Community",
    "description": "Empowering Women in Their Tech Careers",
    "images": [
      {
        "path": "https://cloudprovider.com/image.png",
        "alt": "There are two women talking during a productive mentoring session",
        "type": "desktop"
      }
    ]
  },
  "programmesSection": {
    "title": "Opportunities and Programmes",
    "description": "Join our community and unlock endless opportunities. Network, find mentors, and access leadership programs. Whether you're aiming to enhance your skills, grow your professional network, or advance your career, we have what you need. We offer a wide range of opportunities to help you achieve your goals.\n",
    "programmes": [
      {
        "name": "Mentorship",
        "link": "/mentorship",
        "icon": "diversity_2"
      },
      {
        "name": "Online and in-person events",
        "link": "/events",
        "icon": "event_1"
      },
      {
        "name": "CV Clinic",
        "link": "/projects/cv_clinic",
        "icon": "icon2"
      },
      {
        "name": "Mock Interview",
        "link": "/projects/mock-interview",
        "icon": "interview"
      }
    ]
  },
  "announcementSection": {
    "title": "Announcements",
    "events": [
      {
        "title": "Mentorship ad-hoc registration is open now",
        "date": "Registration from MAY 6 - MAY 17, 2024",
        "endDate": 20240517,
        "description": "Sign up for this unique opportunity of one-time mentoring sessions with personalised guidance from experienced mentors.",
        "link": {
          "label": "Register now",
          "uri": "/mentorship/register-mentee"
        },
        "images": [
          {
            "path": "https://cloudprovider.com/image.png",
            "alt": "There are two women talking during a productive mentoring session",
            "type": "desktop"
          }
        ]
      }
    ]
  },
  "eventsSection": {
    "title": "Events",
    "link": {
      "label": "View all events",
      "uri": "/events"
    },
    "events": [
      {
        "type": "Book Club",
        "date": "THU, MAY 30, 2024, 8:00 PM to 9:30 PM CEST",
        "endDate": 20240517,
        "title": "Book Club: The Pragmatic Programmer\n",
        "host": "Sasha Northon and Lana Rey",
        "speaker": "John Smith",
        "description": "Join us for a discussion of this essential guide to writing clear and maintainable code!\n",
        "images": [
          {
            "path": "https://cloudprovider.com/image.png",
            "alt": "There are two women talking during a productive mentoring session",
            "type": "desktop"
          }
        ],
        "link": {
          "label": "Go to Meetup Event",
          "uri": "https://meetup.com/event1"
        }
      },
      {
        "type": "Tech Talk",
        "date": "THU, JUN 30, 2024, 8:00 PM to 9:30 PM CEST",
        "endDate": 20240517,
        "title": "Get start with Terraform",
        "host": "Airat",
        "speaker": "Sonali",
        "description": "Join us for a discussion of this essential guide to writing clear and maintainable code!\n",
        "images": [
          {
            "path": "https://cloudprovider.com/image.png",
            "alt": "There are two women talking during a productive mentoring session",
            "type": "desktop"
          }
        ],
        "link": {
          "label": "Go to Meetup Event",
          "uri": "https://meetup.com/event1"
        }
      }
    ]
  },
  "volunteerSection": {
    "title": "Volunteer With Us",
    "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
    "link": {
      "label": "Learn more about volunteer",
      "uri": "/about-us/volunteer"
    },
    "images": [
      {
        "path": "https://cloudprovider.com/image.png",
        "alt": "There are two women talking during a productive mentoring session",
        "type": "desktop"
      }
    ]
  }
}
```

File: src/utils/types.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    events: Event[];
+  };
+  announcementSection: {
+    title: string;
+    events: Event[];
+  };
+  volunteerSection: {
+    title: string;
+    description: string;
+    link: string;
```
Comment (dricazenck) [inline src/utils/types.ts:53]: I guess the link here also need to be another type, right?

File: public/home.json (hunk 5 above)
Comment (stepsen89) [reply public/home.json:82]: thanks! I used the one we have in resources in the main branch 👍 updated with this one

File: src/utils/types.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    description: string;
+    images: Image[];
+  };
+  programmesSection: {
+    title: string;
+    description: string;
+    programmes: Programme[];
+  };
+  eventsSection: {
+    title: string;
```
[the remark this answers is not in the document]
Comment (stepsen89) [reply src/utils/types.ts:43]: updated this for all links, and set title as optional as this is not always in the example json

[the remark this answers is not in the document]
Comment (stepsen89) [reply src/utils/types.ts:43]: also btw: updated the README to contain the instructions for env.local now

File: src/utils/types.ts (hunk 6 above)
Comment (dricazenck) [reply src/utils/types.ts:53]:
```suggestion
    link: Link;
```

**Review by dricazenck — APPROVED** [review]

**Pull request merged**